src/main/java/org/opensearch/security/dlic/rest/validation/RequestContentValidator.java

```diff
@@ -0,0 +1,382 @@
+
+    protected final ValidationContext validationContext;
+
+    protected final Map<String, String> wrongDataTypes = new HashMap<>();
+
+    /** Contain errorneous keys */
+    private final Set<String> missingMandatoryKeys = new HashSet<>();
+
+    private final Set<String> invalidKeys = new HashSet<>();
+
+    private final Set<String> missingMandatoryOrKeys = new HashSet<>();
+
+    protected RequestContentValidator(final ValidationContext validationContext) {
+        this.validationError = ValidationError.NONE;
+        this.validationContext = validationContext;
+    }
+
+    public ValidationResult validate(final RestRequest request) throws IOException {
+        return parseRequestContent(request).map(this::validateContentSize).map(jsonContent -> validate(request, jsonContent));
+    }
+
+    public ValidationResult validate(final RestRequest request, final JsonNode jsonContent) throws IOException {
+        return validateContentSize(jsonContent).map(this::validateJsonKeys)
+            .map(this::validateDataType)
+            .map(this::nullValuesInArrayValidator)
+            .map(ignored -> validatePassword(request, jsonContent));
+    }
+
+    private ValidationResult parseRequestContent(final RestRequest request) {
+        try {
+            final JsonNode jsonContent = DefaultObjectMapper.readTree(request.content().utf8ToString());
+            return ValidationResult.success(jsonContent);
+        } catch (final IOException ioe) {
+            LOGGER.error(ValidationError.BODY_NOT_PARSEABLE.message(), ioe);
+            this.validationError = ValidationError.BODY_NOT_PARSEABLE;
+            return ValidationResult.error(this);
+        }
+    }
+
+    private ValidationResult validateContentSize(final JsonNode jsonContent) {
+        if (jsonContent.size() == 0) {
+            this.validationError = ValidationError.PAYLOAD_MANDATORY;
+            return ValidationResult.error(this);
+        }
+        return ValidationResult.success(jsonContent);
+    }
+
+    private ValidationResult validateJsonKeys(final JsonNode jsonContent) {
+        final Set<String> requestedKeys = new HashSet<>();
+        jsonContent.fieldNames().forEachRemaining(requestedKeys::add);
+        // mandatory settings, one of ...
+        if (Collections.disjoint(requestedKeys, validationContext.mandatoryOrKeys())) {
+            missingMandatoryOrKeys.addAll(validationContext.mandatoryOrKeys());
+        }
+        final Set<String> mandatory = new HashSet<>(validationContext.mandatoryKeys());
+        mandatory.removeAll(requestedKeys);
+        missingMandatoryKeys.addAll(mandatory);
+
+        final Set<String> allowed = new HashSet<>(validationContext.allowedKeys().keySet());
+        requestedKeys.removeAll(allowed);
+        invalidKeys.addAll(requestedKeys);
+        if (!missingMandatoryKeys.isEmpty() || !invalidKeys.isEmpty() || !missingMandatoryOrKeys.isEmpty()) {
+            this.validationError = ValidationError.INVALID_CONFIGURATION;
```

Comment:
nit: Is CONFIGURATION the right term to use here? wdyt `INVALID_PAYLOAD`?

Would this message be able to be descriptive around what is invalid about the payload when responding to the caller? i.e. This request is missing required keys key1, key2, key3?

Reply:
> nit: Is CONFIGURATION the right term to use here? wdyt `INVALID_PAYLOAD`?
`INVALID_PAYLOAD` looks better agree originally it is `INVALID_CONFIGURATION`.
> Would this message be able to be descriptive around what is invalid about the payload when responding to the caller? i.e. This request is missing required keys key1, key2, key3?
yes the response something like:
```json
{
  "status": "error"
  "reason": "",
  "keys": "a,b,c"
}
```